lib/rubocop/cop/lint/duplicated_gem.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # A Gem's requirements should be listed only once in a Gemfile.
+      # @example
+      #   # bad
+      #   gem 'rubocop'
+      #   gem 'rubocop'
+      #
+      #   # bad
+      #   group :development do
+      #     gem 'rubocop'
+      #   end
+      #
+      #   group :test do
+      #     gem 'rubocop'
+      #   end
+      #
+      #   # good
+      #   group :development, :test do
+      #     gem 'rubocop'
+      #   end
+      #
+      #   # good
+      #   gem 'rubocop', groups: [:development, :test]
+      class DuplicatedGem < Cop
+        MSG = 'Gem `%s` requirements already given on line %d ' \
+              'of the Gemfile.'.freeze
+
+        def investigate(processed_source)
+          return unless from_gemfile?(processed_source)
+
+          gem_requirements(processed_source.ast)
+            .each_with_object({}) do |node, gem_lines|
+              required_gem = extract_gem_name(node)
+              check_gem(required_gem, node, gem_lines)
+            end
+        end
+
+        private
+
+        def from_gemfile?(processed_source)
+          File.basename(processed_source.path) == 'Gemfile'
+        end
+
+        def gem_requirements(ast)
```

Comment:
Maybe this should be named `#gem_declarations`? When saying "requirements", it makes me think about the version constraints. 😀